token.go

```diff
@@ -163,11 +163,13 @@ func ValidateIdTokenWithNonce(aud string, idToken string, nonce string) (*SiwaId
 	var decodedSignature []byte
 	decodedSignature, err = base64UrlDecode(parts[2])
 	if err != nil {
-		reason = reason + " signature_base64_decode_failed error:" + err.Error()
-		valid = false
-	} else if !verifyAppleRSA256(parts[0]+"."+parts[1], decodedSignature, jwtHeader.Kid) {
-		reason = reason + " signature_verification_failed"
+		reason = reason + fmt.Sprintf(" signature_base64_decode_failed [%s] ", err)
 		valid = false
+	} else {
+		if err := verifyAppleRSA256(parts[0]+"."+parts[1], decodedSignature, jwtHeader.Kid); err != nil {
+			reason = reason + fmt.Sprintf(" signature_verification_failed [%s] ", err)
```

Comment:
nitpick: pull the reason into the Sprintf

Reply:
Good catch. At some point maybe I'll go through the whole repo and clean things up, the Go style is pretty idiosyncratic.